Electron gui generation rapport profiles

abouhana opened the pull request:
Fin mission de génération de rapport de Validation.

abouhana committed:
Create ChangeLog

Ajout d'un ficher de log des modification apportées à Valexa.
Description mission dans ce fichier.

abouhana committed:
Vue Report + communication IPC

Création de la vue ou seront transmis les profils à ajouter au rapport (vide)
Communication IPC entre la vue "Profiles" qui transmets les profils a ajouter a la vue "Report"

AlexECX committed:
update README.md

Ajout de précision pour l'installation de Valexa

abouhana committed:
Ecriture fichier .tex

Modification du fichier .tex pour inclure les données des profiles récupérés avec la communication IPC" Fonctionne, a completer.

AlexECX committed:
Complement fichier .tex

Création des deux tableaux de calibration et validation avec les informations des profiles
Création d'un fichier .tex qui inclue les fichiers paragraphProfiles generés par le python

AlexECX committed:
Ajout informations onglet Report + download files .tex in a .zip

Ajout des informations sur les profiles selectionnés dans l'onglet "Report"
Modification du ChangeLog.md, plus complet
Ajout de l'ensemble des fichiers pour le bon fonctionnement du projet Overleaf (fichiers .tex + .bib + .png)
Telechargement du fichier .zip du projet complet Overleaf dans le dossier "downloads" du pc
Modification du gitignore pour les fichiers .tex générés dans le dossier profiles
Nettoyage des fichiers et dossiers ou seront écrit les fichiers .tex et .png

abouhana committed:
Update ListParagraphsProfile.tex

AlexECX committed:
Update Vue Profile table Uncertainty

2 colonnes avec le meme nom alors aue deux informations différentes

AlexECX committed:
Generation tex project

Creation projet tex complet. Utilisateur a le choix entre generer un pdf (a faire) et le zip des fichiers .tex du projet.
Le zip ne contiennent pas les templates .tex utilisés.

AlexECX committed:
Creation word

Ajout option de download word, pdf et zip a l'utilisateur sur l'onglet "Report"

abouhana committed:
Modification Vue et generation Zip

Zip des png seulement (pas du word et pdf)
Ajout de warnings sur la vue si pas d'options/de profiles selectionnées

abouhana committed:
erreur index loop

erreur copier/coller avec dataValidation et dataCalibration

abouhana committed:
Erreur LOQmax et logneur tables de données

LOQmax au lieu de LOQmin
et ajout d'une ligne supplémentaire au tableaux (pour comter la ligne de titre)

AlexECX commented on valexa/electron_interface/generateReport.py:
```diff
@@ -0,0 +1,284 @@
+    table.style = 'Table Grid'
+    fillTable(table, ["Parameter", "Value"], [
+        ["LOD", "LOQ Min", "LOQ Max", "Correction", "Average Recovery", "Tolerance", "Acceptance"],
+        [
```
Essaie avec des `f""` strings, te permet de faire qqc comme `f"{data['model_info']['lod']} (some text) {data['model_info']['lod']}"`.

Pourrais aussi assigné `data['model_info']` a une variable pour réduire le volume de code. Avoir `model_info['key']` plutot que `data['model_info']['key']`.

AlexECX commented on valexa/electron_interface/generateReportTEX.py:
```diff
@@ -0,0 +1,227 @@
+            lineTemp.clear()
+
+    else:
+        for d in listData:  # listData = list of dict
```
si tu dois couper une "list comprehension", essaie de le faire sur un mot clé:

`{str(key): str(round(val, 4)) if isinstance(val, float) else str(val) for key, val`
 `in d.items()}`

ou meme l'étendre un peu plus:

`{`
 `str(key): str(round(val, 4)) if isinstance(val, float) else str(val)`
 `for key, val in d.items()`
`}`

AlexECX commented on valexa/electron_interface/generateReportTEX.py:
```diff
@@ -0,0 +1,227 @@
+    commands = {
+        "\\COMPOUNDname": data['compound_name'],
+
+        ### SUMMARY TABLES  ###
```
Meme commentaire à propos des f"" strings et d'assigner `model_info` à une variable.

AlexECX commented on src/views/ReportView.vue:
```diff
@@ -0,0 +1,158 @@
+
+  },
+  data: () => ({
+    typeReport: ["pdf", "word"]
```
Généralement on met `title`, `name`, `components` puis `props` et `data`, les hooks du lifecycle (`mounted`, etc) avec `computed`, `methods` et `watchers` à la fin. `props` et `data` son assez central au composant, les placé assez haut est préférable en général.

abouhana committed:
Utilisation f-string

Moins de code avec utilisation de f-string (au liue de concatenation de string) et ajout variable pour data['model_info']
meilleure lisibilité du code dans generateReportTEX

abouhana committed:
utilisation f-string reportTex

Moins de code avec utilisation de f-string (au liue de concatenation de string) et ajout variable pour data['model_info']

abouhana committed:
suppression erreur f-string

erreur de double string inutile

abouhana committed:
Réarrangement balise script ReportView

mise en ordre des composants script dans l'ordre du lifestyle du composant
